modules/extending_and_customizing.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: default
+title: Blueprint Basics
+permalink: /modules/extending-and-customizaing/
+redirect_from:
+  - /extending_and_customizaing.html
+  - /modules/extending_and_customizaing.html
+sitemap:
+    priority: 0.7
+    lastmod: 2015-12-05T18:40:00-00:00
+---
+
+# <i class="fa fa-cube"></i> Blueprint Basics
+
+JHipster has support for `plugins`, and we call them `blueprints` and `modules`.
+
+Prior to JHipster v7.7.0, `modules` were yeoman generators executed using `yo`, extends `generators-jhipster`s `generator-base` class and register hooks to integrate into JHipster workflow.
+
+As of JHipster v7.7.0, `modules` are `blueprints` with stand-alone generators (not blueprinted) with custom cli.
+So we may refer them as standalone blueprints or just blueprints from now on.
+
+## Basic rules for a JHipster blueprint
+
+JHipster blueprints:
+
+- are NPM packages, and are Yeoman generators.
+- follows an extension of the Yeoman rules listed at [https://yeoman.io/authoring/index.html](https://yeoman.io/authoring/index.html). Instead of being prefixed by `generator-`, it is prefixed by `generator-jhipster-`, and instead of having just the `yeoman-generator` keyword, it must have 2 keywords, `yeoman-generator` and `jhipster-blueprint`.
```

Comment:
```suggestion
- follow an extension of the Yeoman rules listed at [https://yeoman.io/authoring/index.html](https://yeoman.io/authoring/index.html). Instead of being prefixed by `generator-`, are prefixed with `generator-jhipster-`, and instead of having just the `yeoman-generator` keyword, and must have 2 keywords, `yeoman-generator` and `jhipster-blueprint`.
```